src/main/kotlin/com/sourcegraph/cody/agent/protocol/ProtocolTextDocument.kt

```diff
@@ -32,6 +36,15 @@ private constructor(
       return Range(position, position)
     }
 
+    private fun getVisibleRange(editor: Editor): Range {
+      val visibleArea = editor.scrollingModel.visibleArea
+      val startOffset = editor.xyToLogicalPosition(visibleArea.location)
+      val endOffset = editor.xyToLogicalPosition(Point(visibleArea.right, visibleArea.bottom))
```

Comment:
The [javadoc](https://github.com/JetBrains/intellij-community/blob/42524aed5922e910f76c77113a8c6f6893b2803a/platform/editor-ui-api/src/com/intellij/openapi/editor/LogicalPosition.java#L14) for `LogicalPosition` says that it can represent a position which doesn't exist in the file (column number larger than the current line length). Are you sure the server would handle that correctly?

Reply:
Maybe it can represent it, but here you have the opposite situation: you get the `LogicalPosition` based on range of the visible code, so there is no such issue there.